src/components/projects/view/report-performance-modal.svelte

```diff
@@ -11,15 +11,20 @@
 
   export let activity;
   export let application;
+  export let report;
   export let isOpen = false;
 
-  let value = {
+  $: value = report || {
     rating: null,
     content: '',
   };
 
   function submit() {
-    dispatch('submit', { value, application, activity });
+    value = {
```

Comment:
Assigning to a reactive variable, and the assignment also happens to just be `value = {...value}`. I don't understand this.

Reply:
This is to unpack the `rating` and `content` fields of the `report` object from the backend, leaving all other fields behind. Maybe not the most elegant, but seems to work.

Reply:
I don't see why it cannot be inlined or even given another name. My guess as to why it works is that reactivity doesn't kick in immediately, but gets scheduled in the event loop which doesn't interrupt this function, so it dispatches with the modified value (which will get reset to the reactive value after this function). Also, the same comment applies to the assignment after dispatching